### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by jcantrill**
Update the README and hacking documenation:
* Update dependencies
* Describe `make run`

### CODE REVIEW

**Comment by jcantrill** [comment]
cc @bparees documentation changes

**Comment by bparees** [comment]
bz override, dev doc changes

File: HACKING.md [hunk 1]
```diff
@@ -1,5 +1,44 @@
...
+**Note:** 
+
+* Modify the log level by `export LOG_LEVEL=debug`
+* CVO and OLM can be scaled back up like `make scale-cvo REPLICAS=1`
+
+
+## Testing
+## Unit Tests
+```
+make test-uniit
```
Comment (periklis) [inline HACKING.md:30]: Typo: `make test-unit`

File: HACKING.md [hunk 2]
```diff
@@ -1,5 +1,44 @@
...
+make deploy
+make test-e2e-olm
+```
+
+**Note:** To skip cleanup of resources while hacking/debugging an E2E test apply `DO_CLEANUP=false`.
 
 ## Updating Dependencies
 Bumping the release and manifest versions typically require udating the `elasticsearch-operator`
-* `dep ensure -update github.com/openshift/elasticsearch-operator`
+* `go get github.com/openshift/elasticsearch-operator`
```
Comment (periklis) [inline HACKING.md:44]: `go get -u`

**Comment by jcantrill** [comment]
/hold cancel

@alanconway please review

**Review by alanconway — COMMENTED** [review]
/lgtm

**Comment by alanconway** [comment]
@jcantrill It's a classic!
```
 Makefile:83: *** missing separator (did you mean TAB instead of 8 spaces?).  Stop. 
```
make is tab-sensitive, every line in a make recipe must start with a tab. 
Looks like github diff ignores whitespace changes so you can't see it in review!

**Comment by alanconway** [comment]
@jcantrill this is a real build error: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-logging-operator/499/pull-ci-openshift-cluster-logging-operator-master-e2e-operator/1321864090515148800#1:build-log.txt%3A108tt

Typo in the makefile: 

   Makefile:83: *** missing separator (did you mean TAB instead of 8 spaces?).  Stop. 

/hold

**Pull request merged**